server/src/actors/project.rs

```diff
@@ -193,27 +192,21 @@ pub struct ProjectConfig {
     pub trusted_relays: Vec<PublicKey>,
     /// Configuration for PII stripping.
     pub pii_config: Option<PiiConfig>,
-    /// List with the fields to be excluded.
-    #[serde(skip_serializing_if = "Vec::is_empty")]
-    pub exclude_fields: Vec<String>,
     /// The grouping configuration.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub grouping_config: Option<Value>,
-    /// Toggles all data scrubbing on or off.
-    #[serde(skip_serializing_if = "is_flag_default")]
-    pub scrub_data: bool,
-    /// Should ip addresses be scrubbed from messages?
-    #[serde(skip_serializing_if = "is_flag_default")]
-    pub scrub_ip_addresses: bool,
-    /// List of sensitive fields to be scrubbed from the messages.
-    #[serde(skip_serializing_if = "Vec::is_empty")]
-    pub sensitive_fields: Vec<String>,
-    /// Controls whether default fields will be scrubbed.
-    #[serde(skip_serializing_if = "is_flag_default")]
-    pub scrub_defaults: bool,
     /// Configuration for filter rules.
     #[serde(skip_serializing_if = "FiltersConfig::is_empty")]
     pub filter_settings: FiltersConfig,
+    /// Configuration for data scrubbers.
+    #[serde(skip_serializing_if = "DataScrubbingConfig::is_disabled")]
+    pub datascrubbing_settings: DataScrubbingConfig,
+
+    /// PII config derived from datascrubbing settings.
+    ///
+    /// Cached here because the conversion process is expensive.
+    #[serde(skip, default = "AtomicLazyCell::new")]
+    datascrubbing_pii_config: AtomicLazyCell<Option<PiiConfig>>,
```

Comment:
Please hide this inside `DataScrubbingConfig` to streamline it with how `GlobPatterns` works internally. Also, this does not need to be an option, since `PiiConfig` has an empty default.

Another benefit of this is that you don't have to add `lazycell`, expose `to_pii_config` and worry about converting at the usage site.